elastic/beats
# New kubernetes module dashboards

## What does this PR do?


This PR modifies existing ones and add more out of the box kubernetes module dashboards as part of https://github.com/elastic/beats/issues/31021

## Why is it important?



The Kubernetes dashboards offered so far where using some fields the wrong way and also where missing a lot of important visualisations.

## Checklist




## Author's Checklist



## How to test this PR locally



- Create a kubernetes cluster with kind
- Bring the stack with `elastic-package stack up --version=8.3.0-SNAPSHOT -v -d `
- Run Metricbeat with image  `8.3.0-SNAPSHOT` in kubernetes https://www.elastic.co/guide/en/beats/metricbeat/current/running-on-kubernetes.html
-  Create metricbeat-* data view in Kibana
- Locally switch to the PR's branch and build Metricbeat with `mage build `
- Setup the dashboards with `./metricbeat setup --dashboards -E setup.dashboards.directory=./module/kubernetes/_meta/kibana/ -E setup.kibana.host=http://elastic:changeme@localhost:5601 -e  -d "*"`
- Check the dashboards in Kibana
- For the shake of this testing would be nice that the k8s cluster has some pods,deployments, statefulsets, daemonsets, jobs, cronjobs, pv, pvc, services,namespaces


## Related issues


- 

## Use cases



## Screenshots



Dashboards added:
<img width="1744" alt="overview-1" src="https://user-images.githubusercontent.com/26270880/170064670-e1e99c16-2793-43da-afef-f1e72612b134.png">
<img width="1756" alt="overview-2" src="https://user-images.githubusercontent.com/26270880/170064681-23022c7c-88a0-4268-9d7d-8600dc283e06.png">
<img width="1665" alt="dashboards pods" src="https://user-images.githubusercontent.com/26270880/167841178-f87e6dd6-6bec-4b3d-a35d-3f6329fa6a74.png">
<img width="1675" alt="pods-1" src="https://user-images.githubusercontent.com/26270880/170065161-303f0bd9-86c5-4207-92b2-bfe588ca90f6.png">
<img width="1748" alt="dashboards nodes" src="https://user-images.githubusercontent.com/26270880/167841234-9b571b7b-483b-49e7-8ac8-6dbe93199168.png">
<img width="1745" alt="dashboards deployments" src="https://user-images.githubusercontent.com/26270880/167841290-47dd85f7-9895-498b-9855-7e2a90693cd1.png">
<img width="1739" alt="dashboards  statefulset" src="https://user-images.githubusercontent.com/26270880/167841345-a3ada135-e706-4e9c-b947-d06ce70b250a.png">
<img width="1740" alt="dashboards volumes" src="https://user-images.githubusercontent.com/26270880/167841442-71030241-f109-41d7-a5ba-1e303fd96d24.png">
<img width="1745" alt="dashboards jobs" src="https://user-images.githubusercontent.com/26270880/167841500-5d9ed96b-1951-48da-9686-e6a66a738f39.png">
<img width="1759" alt="dashboards cronjobs" src="https://user-images.githubusercontent.com/26270880/167841581-5dc14108-2504-48d2-8015-5a919df6db87.png">
<img width="1761" alt="dashboards services" src="https://user-images.githubusercontent.com/26270880/167841681-16ba3489-853c-4f07-9012-2af7b7ce0d2a.png">
<img width="1751" alt="dashboards PVC" src="https://user-images.githubusercontent.com/26270880/167841814-e01c30d8-d6d8-4cac-bf07-cda28e658acb.png">
<img width="1749" alt="dashboards PV:PVC" src="https://user-images.githubusercontent.com/26270880/167841884-b4a096eb-d817-4b67-8421-7964504aac57.png">



## Logs
